android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -676,4 +675,77 @@ private void sendEvent(
         .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
         .emit(eventName, params);
   }
+
+  @ReactMethod
+  public void isSubscribed(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchaseHistoryAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        for (int i = 0; i < list.size(); i++) {
+          PurchaseHistoryRecord purchase = list.get(i);
+          if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+            promise.resolve(true);
+            return;
+          }
+        }
+        promise.resolve(false);
+      });
+    });
+  }
+
+
+  @ReactMethod
+  public void getSubscriptionTransactionDetails(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchasesAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        Purchase purchase = list.get(0);
+        if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+          item.putString("productId", purchase.getSkus().get(0));
+          item.putString("orderId", purchase.getOrderId());
+          item.putString("purchaseToken", purchase.getPurchaseToken());
+          item.putString("purchaseTime", String.valueOf(purchase.getPurchaseTime()));
+          item.putString("transactionReceipt", purchase.getOriginalJson());
+          item.putString("purchaseToken", purchase.getPurchaseToken());
+          item.putString("dataAndroid", purchase.getOriginalJson());
+          item.putString("signatureAndroid", purchase.getSignature());
+        }
+        promise.resolve(item);
+
+      });
+
+    });
+
+  }
```

Comment:
do we expect something to be returned if the user doesn't have any subscriptions? maybe a `promise.resolve(null)` ?